src/apps/chifra/pkg/articulate/trace.go

```diff
@@ -0,0 +1,51 @@
+package articulate
+
+import (
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/abi"
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/types"
+	goEthAbi "github.com/ethereum/go-ethereum/accounts/abi"
+)
+
+func ArticulateTrace(trace *types.SimpleTrace, abiMap abi.AbiInterfaceMap) (articulated *types.SimpleFunction, err error) {
```

Comment:
This may one day be `func (trace *types.SimpleTrace) Articulate(abiMap, abi.AbiInterfaceMap)...`